firedrake/function.py

```diff
@@ -504,9 +504,12 @@ def at(self, arg, *args, **kwargs):
         if not arg.shape:
             arg = arg.reshape(-1)
 
+        mesh = self.function_space().mesh()
+        if mesh.variable_layers:
+            raise NotImplementedError("Point evaluation not implemented for variable layers")
```

Comment:
To do this requires some much more serious refactoring of the point evaluation code.